live-teaching-functions/functions/src/index.ts

```diff
@@ -80,9 +80,24 @@ export const endLiveSession = functions.https.onRequest(async (req, res) => {
       const result = await liveSessionService.endLiveSession(
         req?.body?.data?.live_session_id
       )
-      res.send({data: { status: 'ok', data: result }})
+      res.send({ data: { status: 'ok', data: result } })
     } catch (error) {
-      res.send({data: { status: 'fail', reason: error.message }})
+      res.send({ data: { status: 'fail', reason: error.message } })
+    }
+  })
+})
+
+export const createVoiceClip = functions.https.onRequest(async (req, res) => {
+  corsHandler(req, res, async () => {
+    try {
+      const result = await liveSessionService.createVoiceClip(
+        req?.body?.data?.live_session_id,
+        req?.body?.data?.code,
+        req?.body?.data?.file_URL
+      )
+      res.send({ status: 'ok', data: { ...result, valid: true } })
```

Comment:
If status outer data field that meaning when receiving data from a client you can't see status field.